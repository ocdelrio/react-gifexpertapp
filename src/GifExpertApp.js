import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';



export const GifExpertApp = ({ defaultCategories = []}) => {
    const [catergories, setCategories] = useState(defaultCategories);

    return (
        <>
        <h2>GifExpertApp</h2>
        <AddCategory setCategories={setCategories}/>
        <hr/>

        

        <ol>
            {
                catergories.map( category => 
                    <GifGrid
                        key={category}
                        category={category}/>)
            }
        </ol>
        </>
    )
};
