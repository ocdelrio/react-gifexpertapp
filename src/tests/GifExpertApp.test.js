import React from 'react';
import {shallow} from 'enzyme';
import '@testing-library/jest-dom'
import { GifExpertApp } from '../GifExpertApp';

describe('Test GifExpertApp', () => {

    beforeEach(() => {
        jest.clearAllMocks();
    })

    test('Debe de mostrarse correctamente', () => {
        const wrapper = shallow(<GifExpertApp/>)
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostar una lista de categorias', () => {
        const categories = ['Primera', 'segunda'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>)
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
    

});
