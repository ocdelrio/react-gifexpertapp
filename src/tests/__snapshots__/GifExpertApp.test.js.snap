// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GifExpertApp Debe de mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
